lib/kennel/syncer/matched_expected.rb

```diff
@@ -36,7 +36,12 @@ def matching_expected_lookup_map(expected)
 
         def matching_expected(a, map)
           klass = a.fetch(:klass)
-          map["#{klass.api_resource}:#{a.fetch(:id)}"] || map[a.fetch(:tracking_id)]
+
+          if (e = map[a.fetch(:tracking_id)])
+            return e if e.class.api_resource == klass.api_resource
```

Comment:
can we raise on else ... because that seems broken ...

Reply:
What would be the benefit of raising?

Reply:
I'm hoping it would either make a bad PR fail instantly or show exactly what is broken, since I hope this is from a code bug and not someone copy-pasting tracking ids into different items.

With "self heal" you mean kennel would delete on of the items ?
... that might be alright too, since duplicate ids should fail when generating already

... so either way works 👍